lib/phoenix/transports/long_poll.ex

```diff
@@ -62,18 +62,13 @@ defmodule Phoenix.Transports.LongPoll do
   end
 
   @doc false
-  def call(conn, {endpoint, handler, transport}) do
-    {_, opts} = handler.__transport__(transport)
+  def call(conn, {endpoint, socket_handler, transport}) do
+    {_, opts} = socket_handler.__transport__(transport)
 
     conn
-    |> code_reload(opts, endpoint)
-    |> fetch_query_params
+    |> Transports.Utils.init_plug_conn(endpoint, socket_handler, transport, &status_json(&1, %{}))
```

Comment:
I understand why you did it but I would like to avoid coupling. `init_plug_conn` does nothing and everything at the same time and I think it makes more sense to call the individual functions one by one.